polaris-core/src/main/java/io/polaris/core/PolarisConfiguration.java

```diff
@@ -147,4 +149,36 @@ public static <T> Builder<T> builder() {
               "If set to true, allows tables to have external locations outside the default structure.")
           .defaultValue(false)
           .build();
+
+  public static final PolarisConfiguration<Boolean>
+      ALLOW_TABLE_LOCATION_OUTSIDE_NAMESPACE_LOCATION =
+          PolarisConfiguration.<Boolean>builder()
+              .key("ALLOW_TABLE_LOCATION_OUTSIDE_NAMESPACE_LOCATION")
+              .catalogConfig("allow.table.location.outside.namespace.location")
+              .description(
+                  "If set to true, allows tables to have a location outside of their parent namespace")
+              .defaultValue(false)
+              .build();
+
+  public static final PolarisConfiguration<Boolean> ALLOW_TABLE_LOCATION_OUTSIDE_CATALOG_LOCATION =
+      PolarisConfiguration.<Boolean>builder()
+          .key("ALLOW_TABLE_LOCATION_OUTSIDE_CATALOG_LOCATION")
```

Comment:
This configuration option doesn't make sense to me. The catalog allowed locations specifically dictate where data can live for that catalog. It's also the cornerstone of vended credentials. It seems to me that if users need flexibility in the table locations, the allowed locations set allows for that.

Reply:
One important distinction is that `ALLOWED_LOCATIONS` is a property of the *storage config*, not necessarily of a Catalog itself; a Catalog just happens to hold a storage config. Right now we don't have namespace-level storage configs yet (though we might in the future).

The singular "location" of a catalog or namespace (as opposed to ALLOWED_LOCATIONS) is semantically just the "default-location *hint*".

The options we're talking about with unstructured table location, etc., are just about allowing going off-road from the "default hints", but in general the list of `ALLOWED_LOCATIONS` should always be abided by.